app/controllers/base/ReportGeneratorWizardController.java

```diff
@@ -349,6 +349,26 @@ private String storedFilename(T data) {
 
     protected abstract String documentTableName();
 
+    protected String asLoggableLine(Map<String, String> params) {
+        val keysToKeep = paramsToBeLogged();
+        return params
+                .keySet()
+                .stream()
+                .filter(keysToKeep::contains)
+                .filter(key -> params.get(key) != null)
+                .collect(Collectors.toMap(
+                        Function.identity(),
+                        params::get,
+                        (v1, v2) -> { throw new RuntimeException(String.format("Duplicate key for values %s and %s", v1, v2));},
+                        TreeMap::new))
```

Comment:
Create a sorted map based on key just so log order is always the same, makes testing easier.